packages/@aws-cdk/integ-tests-alpha/lib/assertions/providers/lambda-handler/utils.ts

```diff
@@ -12,11 +12,34 @@ export function parseJsonPayload(payload: any): any {
 export function decodeParameters(obj: Record<string, string>): any {
   return Object.fromEntries(Object.entries(obj).map(([key, value]) => {
     try {
-      return [key, JSON.parse(value)];
+      return [key, decodeValue(value)];
     } catch {
       // if the value cannot be parsed, leave it unchanged
       // this will end up as a string
       return [key, value];
     }
   }));
 }
+
+function decodeValue(value: string): any {
+  const parsed = JSON.parse(value);
+
+  if (parsed != null && !Array.isArray(parsed) && typeof parsed === 'object') {
+    const entries = Object.entries(parsed);
+    if (isByteArray(entries)) {
```

Comment:
But what if the payload is a dictionary that *looks* like a stringified bytearray, but is actually supposed to be a normal dictionary? 

I agree it might be a little edgy, but can we think of a different solution that considers it? Like adding a marker property when we encode, so we can use it when we decode? 

@mrgrain remember this? can you think of another way?

Reply:
I don't even understand how we end up here. Where is that `Uint8Array` coming from, and how did it end up `JSON.stringify()`ed here?

Reply:
It is being passed directly by the user.

```ts
const kinesisStream: IStream = new Stream(stack, "TestKinesisStream");

 const buf = Buffer.from(JSON.stringify({ name: "world" }));
 const data = new Uint8Array(
   buf.buffer,
   buf.byteOffset,
   buf.byteLength / Uint8Array.BYTES_PER_ELEMENT
 );

 testCase.assertions.awsApiCall("Kinesis", "putRecord", {
   StreamName: kinesisStream.streamName,
   Data: data,
   PartitionKey: "testKey",
 });
```

Reply:
Yeah, there is a reason why untyped arguments that go over the network should really be JSON strings 🙈 
I wonder if this is even a SDKv3 bug. I know we changed the handling here, but this might have not worked before (but I don't know).

I'll guess we have to put a marker on it if we want to support it. Also should probably base64 encode it during transit.

Reply:
Yeah a `Uint8Array` cannot be expected to be sent over the wire just like that.

Either:
- We reject the value, but allow the use case by making the equality testing a bit smarter in the assertions library -- perhaps we can test the equality of a Uint8Array to a string (assuming utf-8 encoding), or an array of numbers.
- Either that, or we marshal the `Uint8Array` explicitly over the wire with a type tag, and reconstruct it on the other end.

Honestly, I'm feeling most for the first solution. Morally, what we are trying to do is test equality of the buffer to a certain string. The fact that we need to encode/decode that string/byte array could be an implementation detail that that the assertions module just takes care of.